src/mlpack/core/data/tokenizer.hpp

```diff
@@ -0,0 +1,64 @@
+/**
+ * @file tokenizer.hpp
+ * @author Jeffin Sam
+ *
+ * Implementation of Tokenizer class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_TOKENIZER_HPP
+#define MLPACK_CORE_DATA_TOKENIZER_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace data {
+/**
+ * A simple Tokenizer class
+ */
+class Tokenizer
```

Comment:
I suggest to rename this class since it's not an abstract tokenizer. This class splits a string using a set of one-charecter delimiters.

Reply:
used `strtok` as class name analogous to c `strtok()`